Dockerfile

```diff
@@ -8,23 +8,89 @@
 # license agreement from NVIDIA CORPORATION is strictly prohibited.
 #
 
-ARG CUDA_VERSION=9.2
-ARG LINUX_VERSION=ubuntu16.04
-ARG CC_VERSION=5
-ARG CXX_VERSION=5
+ARG FROM_IMAGE=gpuci/miniconda-cuda
+ARG CUDA_VER=10.1 
+ARG LINUX_VERSION=ubuntu18.04
+ARG CC_VERSION=7
+ARG IMAGE_TYPE=devel
 ARG PYTHON_VERSION=3.6
 
-FROM gpuci/rapidsai-base:cuda${CUDA_VERSION}-${LINUX_VERSION}-gcc${CC_VERSION}-py${PYTHON_VERSION}
+FROM ${FROM_IMAGE}:${CUDA_VER}-${IMAGE_TYPE}-${LINUX_VERSION}
 
-# Install htslib dependencies
-RUN apt-get update
-RUN apt-get install -y tabix \
-        zlib1g-dev \
-        libbz2-dev \
-        liblzma-dev \
-        libcurl4-gnutls-dev \
-        wget \
-        libssl-dev      # VariantWorks `cyvcf2` dependency
+# Capture argument used for FROM
+ARG CC_VERSION
+ARG PYTHON_VERSION
+
+# Update environment for gcc/g++ builds
+ENV CC=/usr/bin/gcc
+ENV CXX=/usr/bin/g++
+ENV CUDAHOSTCXX=/usr/bin/g++
+ENV CUDA_HOME=/usr/local/cuda
+ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/cuda/lib64:/usr/local/lib
+
+# Enables "source activate conda"
+SHELL ["/bin/bash", "-c"]
+
+RUN apt update && \
+    apt-get -y install  vim \
+    # Install gcc version
+    g++-${CC_VERSION} \
+    gcc-${CC_VERSION} \
+    cmake \
+    make \
+    # Install the packages needed to build with.
+    # Install htslib dependencies
+    tabix \
+    zlib1g-dev \
+    libbz2-dev \
+    liblzma-dev \
+    libcurl4-gnutls-dev \
+    wget \
+    # VariantWorks `cyvcf2` dependency
+    libssl-dev \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+    
+RUN update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-${CC_VERSION} 1000 --slave /usr/bin/g++ g++ /usr/bin/g++-${CC_VERSION}
+
+# Add a condarc for channels and override settings
+RUN echo -e "\
+ssl_verify: False \n\
+channels: \n\
+  - gpuci \n\
+  - conda-forge \n\
+  - nvidia \n\
+  - defaults \n" > /conda/.condarc \
+      && cat /conda/.condarc ;
+
+# Create rapids conda env and make default
+RUN source activate base \
+    && conda install -y gpuci-tools \
+    && gpuci_conda_retry create --no-default-packages --override-channels -n rapids \
+      -c nvidia \
+      -c conda-forge \
+      -c defaults \
+      -c gpuci \
+      git \
+      gpuci-tools \
+      python=${PYTHON_VERSION} \
+      "setuptools<50" \
+    && sed -i 's/conda activate base/conda activate rapids/g' ~/.bashrc ;
 
 # Install htslib
-RUN wget https://github.com/samtools/htslib/releases/download/1.9/htslib-1.9.tar.bz2 && tar xvf htslib-1.9.tar.bz2 && cd htslib-1.9 && ./configure && make -j16 install
+RUN wget https://github.com/samtools/htslib/releases/download/1.9/htslib-1.9.tar.bz2 \
```

Comment:
we're about to add `htslib` as a submodule into GenomeWorks, so I think we can skip installing this from source. We could add the `apt` install for this instead. I think the package name is `libhts-dev`

Reply:
Alright, I made this change for Ubuntu. For centOS I think this would be better to build it from source since it requires downloading 3 rpm's from https://centos.pkgs.org/7/epel-x86_64/htslib-devel-1.9-5.el7.x86_64.rpm.html

Reply:
hmm then it might be better to revert it, or maybe even install through `conda`. best to have the same version across ubuntu and centos to avoid version related bugs

Reply:
Done. Moved the installation of `htslib` to Anaconda for both

Reply:
awesome, thank you!